baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/UnusedParameter.java

```diff
@@ -0,0 +1,939 @@
+import com.sun.source.tree.LambdaExpressionTree;
+import com.sun.source.tree.MemberReferenceTree;
+import com.sun.source.tree.MemberSelectTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.ReturnTree;
+import com.sun.source.tree.StatementTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.TryTree;
+import com.sun.source.tree.UnaryTree;
+import com.sun.source.tree.VariableTree;
+import com.sun.source.tree.WhileLoopTree;
+import com.sun.source.util.SimpleTreeVisitor;
+import com.sun.source.util.TreePath;
+import com.sun.source.util.TreePathScanner;
+import com.sun.source.util.TreeScanner;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Type;
+import com.sun.tools.javac.tree.JCTree;
+import com.sun.tools.javac.util.Position;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import javax.annotation.Nullable;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.Name;
+
+/**
+ * Copy from {@Link UnusedVariable } except we consider all parameter for unused analysis.
+ * We removed the `onlyCheckForReassignments` filter
+ * */
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "UnusedParameter",
+        altNames = {"unused", "UnusedParameters"},
+        summary = "Unused.",
+        providesFix = REQUIRES_HUMAN_ATTENTION,
+        severity = WARNING,
+        documentSuppression = false)
+public final class UnusedParameter extends BugChecker implements BugChecker.CompilationUnitTreeMatcher {
+    private static final String EXEMPT_PREFIX = "unused";
+
+    /**
+     * The set of annotation full names which exempt annotated element from being reported as unused.
+     */
+    private static final ImmutableSet<String> EXEMPTING_VARIABLE_ANNOTATIONS =
+            ImmutableSet.of(
+                    "javax.persistence.Basic",
+                    "javax.persistence.Column",
+                    "javax.persistence.Id",
+                    "javax.persistence.Version",
+                    "javax.xml.bind.annotation.XmlElement",
+                    "org.junit.Rule",
+                    "org.mockito.Mock",
+                    "org.openqa.selenium.support.FindBy",
+                    "org.openqa.selenium.support.FindBys");
+
+    /** The set of types exempting a type that is extending or implementing them. */
+    private static final ImmutableSet<String> EXEMPTING_SUPER_TYPES =
+            ImmutableSet.of(
+            );
+
+    /** The set of types exempting a field of type extending them. */
+    private static final ImmutableSet<String> EXEMPTING_FIELD_SUPER_TYPES =
+            ImmutableSet.of("org.junit.rules.TestRule");
```

Comment:
The rule will ignore any parameter prefixed with `unused` so you should be able to easily avoid getting hit.

In general, I'd like to avoid adding special casing so that its easy to keep the rule up to date